test/dense.jl

```diff
@@ -69,8 +69,8 @@ fill!(B, -1)
 @test badcopy!(B, A) == A
 fill!(C, -1)
 badcopy!(C, A)
-@test C[2,1] != A[2,1]   # oops!
-@test C[2,1] == A[1,2]
+@test C.parent[2,1] != A[2,1]   # oops!
```

Comment:
Not sure what's the point of this test. It was giving an error because `badcopy!` was actually doing the right thing. But when I used `C.parent` instead of `C`, the tests were passed because `parent` had the matrix transposed.

Reply:
It's been ages since I worked on this (changes in 0.7 might make this viable again), but if memory serves then I think the `eachindex` definition in `test/array_types.jl` should be returning a `PDAIterator` which should (deliberately) do the wrong thing for a `PermutedDimsArray`.